enhancements/cloud-integration/infrastructure-external-platform-type.md

```diff
@@ -0,0 +1,338 @@
+
+Having special, built-in and somewhat generic, platform-type which will signal about presence of underlying infrastructure platform without
+platform specific details might help to reduce a number of pull-requests across OCP repositories
+and simplify initial integration work for external to RedHat contributors.
+
+In the future, at some degree, [capabilites selection](https://github.com/openshift/enhancements/blob/master/enhancements/installer/component-selection.md)
+API might help to solve the issue and give a room for supplementing *some* platform dependant components, however some integral parts of 
+OpenShift can not be disabled and still requires a signal about underlying platform, for example KCM and MCO in respect of enablement of
+[external cloud controller manager](https://kubernetes.io/docs/concepts/architecture/cloud-controller/)
+
+Examples of difficulties which presents now due to predefined list of platforms:
+- No defined mechanism to set `--cloud-provider=external` arg to kubelet/KCM/apiserver without merges and further revendoring of "openshift/api"
+- No way to extend machine-api and deliver new provider without merges to "openshift/machine-api-operator" and "openshift/api" repos
+
+### User Stories
+
+- As a cloud-provider affiliated engineer / platform integrator / RH partner
+  I want to have a mechanism to signal Openshift's built-in operators about additional
+  cloud-provider specific components which should be delivered and managed by its own or manually.
+  
+  We are aware about examples of supplementing OKD installations with a custom machine-api controllers,
+  however users are experiencing a lot of difficulties on this path due to necessity of, literally, reverse engineering,
+  manual management of generic MAPI controllers and so on.
+
+### Goals
+
+- Reduce the necessity to make changes in "openshift/api", "openshift/library-go" and dependant infrastructure related components during basic cloud-provider integration with OpenShift
+- Make cloud-provider integration process more accessible and simple to external contributors as well as for RH engineers
+- Provide a top level description of the operation mode for affected components at the first steps
+- Introduce a somewhat "neutral" platform type, which would serve as a signal about underlying cloud-provider presence
+
+### Non-Goals
+
+- describe concrete delivery mechanisms for cloud-provider specific components
+- cover new infrastructure provider enablement from RHCOS side
+- describe specific changes for each affected component, aside no-op reaction to thew new "External" platform type
+
+## Proposal
+
+Extend the [PlatformType](https://github.com/openshift/api/blob/51f399230d604fa013c2bb341040c4ad126e7309/config/v1/types_infrastructure.go#L128) "enum"
+with a special value "External". Which would be similar to the "None" value during initial phase, but alongside adding support of this across
```

Comment:
in essence yes, as part of the epic that we put together for this work we defined a 3 phase approach to implementation. the first phase is creating the enhancement and identifying needed capabilities. the second phase will involve making sure that the new platform type does not disrupt the current openshift operation. phase 3 will involve modifications to the operators so that they do specific things when seeing an external platform, this will be highly dependent on what each operator requires.

Reply:
IMO we should describe the intended end state rather than the implementation details at the `Proposal` section of the enhancment. You can talk about the phases and stuff later but the `Proposal` should be a high level what are we going to achieve at the end of this project